main.go

```diff
@@ -1,22 +1,38 @@
 package main
 
 import (
+	"bufio"
 	"fmt"
+	"io"
 	"log"
 	"os"
+	"strings"
 )
 
 func main() {
-	// fmt.Println("WE ARE SO COOL!!! 🎉")
-	statsInfo, err := os.Stdin.Stat()
 
-	if err != nil {
-		log.Fatal("OH NOES THIS IS SO BROKEN!!")
-	}
+	reader := bufio.NewReader(os.Stdin)
+
+	for {
+		line, err := reader.ReadString('\n')
+
+		// if end of file, just break, else shout
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			log.Fatal("OH NOES THIS IS SO BROKEN!!")
 
-	fmt.Printf("%#v", statsInfo)
+		}
+
+		// reader.ReadString() retains delimiter, so this strips it for output
+		line = strings.TrimSuffix(line, "\n")
+
+		fmt.Println(line)
+	}
 
-	if statsInfo.Mode()&os.ModeNamedPipe == 0 {
+	if os.ModeNamedPipe == 0 {
```

Comment:
What you had before here was, I think, correct. I can show you why that's true.

Should also do it above in my opinion but I think we'll do some organizational things today so no big deal really!